### PR TITLE
Feature/bootstrap

### PR DESCRIPTION
**Pull request by luckychess**
Discovering tool for Iroha's bootstraping (https://github.com/hyperledger/iroha/issues/254).

### CODE REVIEW

File: tools/make_hostdiscover.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+int main(int argc, char* argv[]) {
+    std::vector<std::string> defaultHosts {};
+    auto trustedHosts = config::IrohaConfigManager::getInstance().getTrustedHosts(defaultHosts);
+    if (trustedHosts.empty()) {
+        std::cout << "Config section for trusted hosts was not found or it is empty." << std::endl;
+        return 1;
+    }
+
+    std::ofstream hostsDiscovered("discovered.txt");
```
Comment (Warchant) [inline tools/make_hostdiscover.cpp]: just print to stdout in format
```
ip pubkey
ip pubkey
```

don't save to file

Comment (luckychess) [inline tools/make_hostdiscover.cpp]: @Warchant stdout is spammed with grpc logs, config read logs and so on:
```
1490863700 DEBUG [config] load json is {
  "database_path":"/tmp/iroha_ledger",
  "java_class_path":"java_tests",
  "java_class_path_local":"smart_contract/java_tests",
  "java_library_path":"lib",
  "java_library_path_local":"build/lib",
  "java_policy_path":"jvm/java.policy.txt",
  "concurrency": 0,
  "max_faulty_peers" : 1,
  "pool_worker_queue_size": 1024,
  "http_port": 1204,
  "grpc_port": 50051,
  "active_start": false,
  "trusted_hosts": [
    "127.0.0.1",
    "172.17.0.2",
    "172.17.0.3",
    "172.17.0.4",
    "172.17.0.5",
    "188.130.155.39/8"
  ]
}

1490863700 INFO [connection] Host discover service
1490863700 INFO [connection] message: discovery
1490863700 INFO [connection] response: IaDYg2mKk2u1JQcKw5PL4qgdDKVKyj+l74rvu7xrS1k=
127.0.0.1 IaDYg2mKk2u1JQcKw5PL4qgdDKVKyj+l74rvu7xrS1k=
1490863700 INFO [connection] Host discover service
1490863700 INFO [connection] message: discovery
1490863701 INFO [connection] Host discover service
1490863701 INFO [connection] message: discovery
1490863702 INFO [connection] Host discover service
1490863702 INFO [connection] message: discovery
1490863703 INFO [connection] Host discover service
1490863703 INFO [connection] message: discovery
1490863704 WARNING [Networks larger than /16 are not supported for now] 
```
So I think file is better.

**Comment by luckychess** [comment]
Now user can pass file name to save discovered nodes as a command line parameter.
@Warchant merge please if you think all is okay.

File: tools/make_hostdiscover.cpp [hunk 2]
```diff
@@ -26,14 +26,24 @@ using Api::DiscoverRequest;
 using Api::Peer;
 
 int main(int argc, char* argv[]) {
+
+    std::string discoveredFileName = "discovered.txt";
+    if (argc == 2) discoveredFileName.assign(argv[1]);
+    if (argc > 2) {
```
Comment (Warchant) [inline tools/make_hostdiscover.cpp]:
```
argc != 2
```

Comment (luckychess) [inline tools/make_hostdiscover.cpp]: After some discussion with @Warchant we decided it's okay.

**Pull request merged**